shiny/ui/_chat.py

```diff
@@ -1325,6 +1376,210 @@ async def _send_custom_message(self, handler: str, obj: ClientMessage | None):
             },
         )
 
+    @overload
+    def enable_bookmarking(
+        self,
+        client: ClientWithState,
+        /,
+        *,
+        bookmark_on: Optional[Literal["response"]] = "response",
+    ) -> CancelCallback: ...
+    @overload
+    def enable_bookmarking(
+        self,
+        client: ChatlasClient[Any, Any],
+        /,
+        *,
+        bookmark_on: Optional[Literal["response"]] = "response",
+    ) -> CancelCallback: ...
+    def enable_bookmarking(
+        self,
+        client: Any,
+        /,
+        *,
+        # TODO: Barret - bookmark_on docs
+        bookmark_on: Optional[Literal["response"]] = "response",
+    ) -> CancelCallback:
+        """
+        Enable bookmarking for the chat instance.
+
+        This method allows the chat instance to be bookmarked, which means that the
+        current state of the chat (including messages) can be saved and restored
+        when the app is reloaded.
+
+        Parameters
+        ----------
+        client
+            The chat client (e.g. [chatlas](https://posit-dev.github.io/chatlas/)) instance to use for bookmarking.
```

Comment:
```suggestion
            The chat client instance to use for bookmarking. This can be a Chat model provider from [chatlas](https://posit-dev.github.io/chatlas/), or more generally, an instance following the `ClientWithState` protocol
```